js/index.js

```diff
@@ -43,27 +43,129 @@ function renderPepperoni() {
+  const gluten = document.querySelector('.crust');
+  if (state.glutenFreeCrust) {
+    gluten.classList.remove('crust-gluten-free');
+  } else {
+    gluten.classList.add('crust-gluten-free')
+  }
 }
 
+
 function renderButtons() {
   // Iteration 3: add/remove the class "active" of each `<button class="btn">`
-}
+
+    const btnPepperoni = document.querySelector('.btn.btn-pepperoni')
+    if (!state.pepperoni) {
+      btnPepperoni.classList.remove('active')
+    } else {
+      btnPepperoni.classList.add('active')
+    }
+    const btnMushrooms = document.querySelector('.btn.btn-mushrooms')
+    if (!state.mushrooms) {
+      btnMushrooms.classList.remove('active')
+    } else {
+      btnMushrooms.classList.add('active')
+    }
+    btnGreenPeppers = document.querySelector('.btn.btn-green-peppers')
+    if (!state.greenPeppers) {
+      btnGreenPeppers.classList.remove('active')
+    } else {
+      btnGreenPeppers.classList.add('active')
+    }
+    btnWhiteSauce = document.querySelector('.btn.btn-sauce')
+    if (!state.whiteSauce) {
+      btnWhiteSauce.classList.remove('active')
+    } else {
+      btnWhiteSauce.classList.add('active')
+    }
+    btnGlutenFree = document.querySelector('.btn.btn-crust')
+    if (!state.glutenFreeCrust) {
+      btnGlutenFree.classList.remove('active')
+    } else {
+      btnGlutenFree.classList.add('active')
+    }
+  }
+
+  function createIngredientLi(text){
+    var li = document.createElement('li')
+    li.innerText = text
+    return li;
+  }
+
 
 function renderPrice() {
-  // Iteration 4: change the HTML of `<aside class="panel price">`
+  
+  const priceList = document.querySelector('.price ul')
+
+  const totalPrice = document.querySelector('.price strong')
+
+ 
+  
+  var child = priceList.lastElementChild;  
+  while (child) { 
+      priceList.removeChild(child); 
+      child = priceList.lastElementChild; 
+  } 
+
+  let acumulado = 0
+
+  if (state.pepperoni) {
+      priceList.appendChild(createIngredientLi("$" + ingredients.pepperoni.price + " " + ingredients.pepperoni.name))
+      acumulado += ingredients.pepperoni.price
+  }
+
+  if (state.mushrooms) {
+    priceList.appendChild(createIngredientLi("$" + ingredients.mushrooms.price + " " + ingredients.mushrooms.name))
+    acumulado += ingredients.mushrooms.price
+  }
+  if (state.greenPeppers) {
+    priceList.appendChild(createIngredientLi("$" + ingredients.greenPeppers.price + " " + ingredients.greenPeppers.name))
+    acumulado += ingredients.greenPeppers.price
+  }
+  if (state.whiteSauce) {
+    priceList.appendChild(createIngredientLi("$" + ingredients.whiteSauce.price + " " + ingredients.whiteSauce.name))
+    acumulado += ingredients.whiteSauce.price
+  }
+  if (state.glutenFreeCrust) {
+    priceList.appendChild(createIngredientLi("$" + ingredients.glutenFreeCrust.price + " " + ingredients.glutenFreeCrust.name))
+    acumulado += ingredients.glutenFreeCrust.price
+
+  }
```

Comment:
Muy bien! Aquí también puedes evitar redundancias, usando un bucle `for... in` de la siguiente forma:
```
for(let oneIngredient in ingredients) {
    if(state[oneIngredient]) {
       // ...
     } 
}
```